### PR TITLE
precompile

### PR DESCRIPTION
**Pull request by frankyin-factual**
stable asset precompiles

### CODE REVIEW

File: runtime/common/src/precompile/stable_asset.rs [hunk 1]
```diff
@@ -0,0 +1,672 @@
...
+#[module_evm_utility_macro::generate_function_selector]
+#[derive(RuntimeDebug, Eq, PartialEq, TryFromPrimitive, IntoPrimitive)]
+#[repr(u32)]
+pub enum Action {
+	GetStableAssetPoolPoolTokens = "getStableAssetPoolTokens(uint32)",
+	GetStableAssetPoolPoolTotalSupply = "getStableAssetPoolPoolTotalSupply(uint32)",
+	GetStableAssetPoolPoolPrecision = "getStableAssetPoolPoolPrecision(uint32)",
+	GetStableAssetPoolPoolMintFee = "getStableAssetPoolPoolMintFee(uint32)",
+	GetStableAssetPoolPoolSwapFee = "getStableAssetPoolPoolSwapFee(uint32)",
+	GetStableAssetPoolPoolRedeemFee = "getStableAssetPoolPoolRedeemFee(uint32)",
```
Comment (zjb0807) [inline runtime/common/src/precompile/stable_asset.rs:50]:
```suggestion
	GetStableAssetPoolTokens = "getStableAssetPoolTokens(uint32)",
	GetStableAssetPoolTotalSupply = "getStableAssetPoolTotalSupply(uint32)",
```

File: runtime/acala/src/lib.rs [hunk 2]
```diff
@@ -1499,6 +1502,68 @@ impl module_idle_scheduler::Config for Runtime {
...
+	}
+}
+
+pub struct ConvertBalanceHoma;
+impl orml_tokens::ConvertBalance<Balance, Balance> for ConvertBalanceHoma {
+	type AssetId = CurrencyId;
+
+	fn convert_balance(balance: Balance, asset_id: CurrencyId) -> Balance {
+		match asset_id {
+			CurrencyId::Token(TokenSymbol::LKSM) => {
```
Comment (zjb0807) [inline runtime/acala/src/lib.rs:1518]:
```suggestion
			CurrencyId::Token(TokenSymbol::LDOT) => {
```

File: runtime/common/src/precompile/mod.rs [hunk 3]
```diff
@@ -141,6 +144,7 @@ where
 				ORACLE,
 				// SCHEDULER,
 				DEX,
+				STABLE_ASSET,
```
Comment (zjb0807) [inline runtime/common/src/precompile/mod.rs:147]:
```suggestion
```

Comment (zjb0807) [reply runtime/common/src/precompile/mod.rs:147]: Requires a predeployed contract to call the precompile

File: runtime/common/src/precompile/multicurrency.rs [hunk 4]
```diff
@@ -452,7 +452,7 @@ mod tests {
 
 			// 1_000_000_000
```
Comment (zjb0807) [inline runtime/common/src/precompile/multicurrency.rs:453]:
```suggestion
			// 2_000_000_000
```

File: runtime/common/src/precompile/stable_asset.rs [hunk 5]
```diff
@@ -0,0 +1,672 @@
...
+pub struct StableAssetPrecompile<R>(PhantomData<R>);
+
+#[module_evm_utility_macro::generate_function_selector]
+#[derive(RuntimeDebug, Eq, PartialEq, TryFromPrimitive, IntoPrimitive)]
+#[repr(u32)]
+pub enum Action {
+	GetStableAssetPoolPoolTokens = "getStableAssetPoolTokens(uint32)",
+	GetStableAssetPoolPoolTotalSupply = "getStableAssetPoolPoolTotalSupply(uint32)",
+	GetStableAssetPoolPoolPrecision = "getStableAssetPoolPoolPrecision(uint32)",
+	GetStableAssetPoolPoolMintFee = "getStableAssetPoolPoolMintFee(uint32)",
```
Comment (zjb0807) [inline runtime/common/src/precompile/stable_asset.rs:48]: Why are there two Pools here? `PoolPool`

Comment (frankyin-factual) [reply runtime/common/src/precompile/stable_asset.rs:48]: Ok. I didn't realize that was the case. Gonna fix that.

**Review by xlc — APPROVED** [review]

**Review by zjb0807 — APPROVED** [review]

**Pull request merged**